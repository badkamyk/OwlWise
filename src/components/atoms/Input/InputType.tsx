export type InputProps = {
	type: string;
	id: string;
	placeholder: string;
	name?: string
	required?: boolean;
}