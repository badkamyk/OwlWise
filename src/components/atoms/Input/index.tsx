import { InputProps } from "./InputType";

export default function Input({ type, id, name, placeholder, required }: InputProps) {
 return (
	 <input type={type} id={id}
					className="shadow-sm bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-primary-500 focus:border-primary-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-primary-500 dark:focus:border-primary-500 dark:shadow-sm-light"
					placeholder={placeholder} name={name} required={required} />
	 );}