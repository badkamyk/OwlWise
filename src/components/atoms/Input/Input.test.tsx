import { it, expect } from "vitest";
import { render, renderHook, screen } from "@testing-library/react";
import Input from "./index";
import { FormProvider, useForm } from "react-hook-form";


it("Input with props", () => {
	const { result } = renderHook(() => useForm());
	render(
		<FormProvider {...result.current}>
			<Input type="text" id="name" name="name" placeholder="Name" required />
		</FormProvider>,
	);
	expect(result).toBeTruthy();
	expect(screen.findByPlaceholderText("Name")).toBeTruthy();
	expect(result).toMatchSnapshot();
});