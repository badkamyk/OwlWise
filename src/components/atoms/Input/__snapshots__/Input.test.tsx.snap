// Vitest Snapshot v1

exports[`Input with props 1`] = `
{
  "current": {
    "clearErrors": [Function],
    "control": {
      "_defaultValues": {},
      "_executeSchema": [Function],
      "_fields": {
        "name": {
          "_f": {
            "mount": true,
            "name": "name",
            "ref": <input
              aria-invalid="false"
              class="focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-500 dark:focus:border-primary-500 dark:shadow-sm-light block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 shadow-sm dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400"
              id="name"
              name="name"
              placeholder="Name"
              required=""
              type="text"
            />,
          },
        },
      },
      "_focusError": [Function],
      "_formState": {
        "dirtyFields": {},
        "errors": {},
        "isDirty": false,
        "isLoading": true,
        "isSubmitSuccessful": false,
        "isSubmitted": false,
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "submitCount": 0,
        "touchedFields": {},
      },
      "_formValues": {
        "name": "",
      },
      "_getDirty": [Function],
      "_getFieldArray": [Function],
      "_getWatch": [Function],
      "_names": {
        "array": Set {},
        "mount": Set {
          "name",
        },
        "unMount": Set {},
        "watch": Set {},
      },
      "_options": {
        "mode": "onSubmit",
        "reValidateMode": "onChange",
        "shouldFocusError": true,
      },
      "_proxyFormState": {
        "dirtyFields": false,
        "errors": "all",
        "isDirty": false,
        "isValid": false,
        "isValidating": false,
        "touchedFields": false,
      },
      "_removeUnmounted": [Function],
      "_reset": [Function],
      "_stateFlags": {
        "action": false,
        "mount": true,
        "watch": false,
      },
      "_subjects": {
        "array": {
          "next": [Function],
          "observers": [],
          "subscribe": [Function],
          "unsubscribe": [Function],
        },
        "state": {
          "next": [Function],
          "observers": [
            {
              "next": [Function],
            },
          ],
          "subscribe": [Function],
          "unsubscribe": [Function],
        },
        "watch": {
          "next": [Function],
          "observers": [],
          "subscribe": [Function],
          "unsubscribe": [Function],
        },
      },
      "_updateFieldArray": [Function],
      "_updateFormState": [Function],
      "_updateValid": [Function],
      "getFieldState": [Function],
      "register": [Function],
      "unregister": [Function],
    },
    "formState": {
      "defaultValues": undefined,
    },
    "getFieldState": [Function],
    "getValues": [Function],
    "handleSubmit": [Function],
    "register": [Function],
    "reset": [Function],
    "resetField": [Function],
    "setError": [Function],
    "setFocus": [Function],
    "setValue": [Function],
    "trigger": [Function],
    "unregister": [Function],
    "watch": [Function],
  },
}
`;
